tin/tests/tin_test.py

```diff
@@ -0,0 +1,67 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+
+from .create_users import make_users
+
+DEFAULT_STUDENT_USERNAME = "2027adeshpan"
+PASSWORD = "2024kshankar-loves-tin(foil)"
+User = get_user_model()
+
+
+class TinTestCase(TestCase):
+    """
+    A convenience class for common Tin operations in tests,
+    like making users, admins, and staff.
+
+    The default client login is as admin, but can be changed
+    via the ``usertype`` class attribute.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        class MyStudentTest(TinTestCase):
+            # now self.client is a student
+            usertype = "student"
+
+
+        class MyStaffTest(TinTestCase):
+            # now self.client is a staff member only
+            usertype = "staff"
+
+
+        class MyNonLoggedInTest(TinTestCase):
+            # self.client will not be logged in
+            usertype = None
+
+    """
+
+    usertype: str = "admin"
+    """
+    Whether logged in user is an admin, teacher, staff, or student.
```

Comment:
What's `teacher` vs. `staff`?

Reply:
As discussed internally, staff is simply a label allowing a user to access django admin.